wordle-web/app.vue

```diff
@@ -24,23 +50,62 @@
 <script setup lang="ts">
 import { useTheme } from "vuetify";
 import nuxtStorage from "nuxt-storage";
-
+import { provide, ref, watch, onMounted, inject } from "vue";
 const router = useRouter();
 const theme = useTheme();
 const showHelpDialog = ref(false);
+const showUserNameDialog = ref(false);
+const userName = ref("guest");
+
+provide("userNameValue", userName.value);
+provide("userName", userName);
+provide("showUserNameDialog", showUserNameDialog.value);
+
+
+const themes = [
+  { name: "default", theme: "dark" },
+  { name: "sans", theme: "sansDark" },
+  { name: "watermelon", theme: "jarringDark" },
+];
 
+const pages = [
+  { name: "Home", path: "/" },
+  { name: "About", path: "/about" },
+  { name: "Leader Board", path: "/leaderboard" },
+];
 onMounted(() => {
-  var defaultTheme = nuxtStorage.localStorage.getData("theme");
-  theme.global.name.value = defaultTheme ?? "dark";
+  var defaultTheme = nuxtStorage.localStorage.getData('theme');
+  if (defaultTheme) { changeTheme(defaultTheme); }
+
+  var userNameStored = nuxtStorage.localStorage.getData('userName');
+  if (userNameStored) {
+    userName.value = userNameStored;
+    showUserNameDialog.value = false;
+  }
+  if (userName.value === "guest " || !userNameStored) {
+    showUserNameDialog.value = true;
+  }
```

Comment:
Side effect of having this within the `app.vue` is that it will trigger the dialog on other pages like `about` and `leaderboard` for user to enter in username. Consider moving this to index.vue to fix this.

Reply:
Agreed, we should probably just move the username there.

Reply:
This is intentional, I wanted the username to show up in the app bar and also have it as a button to change it on the app bar